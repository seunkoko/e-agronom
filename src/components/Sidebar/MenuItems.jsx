// react library
import React from 'react';
import { Link } from 'react-router-dom';

// styles
import './Sidebar.scss';

/**
 * Renders MenuItems component
 * @param {object} props - component props
 * @returns {JSX} JSX
 */
const MenuItems = ({
  id,
  name
}) => (
  <Link
    to={`/${id}`}
    className="links"
  >
    <div className="menu-items">
      <i className="fas fa-cat menu-cat-icon"></i>
      <p className="menu-item-name">{name}</p>
    </div>
  </Link>
);

export default MenuItems;
