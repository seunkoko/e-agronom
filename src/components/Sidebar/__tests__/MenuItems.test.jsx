// react library
import React from 'react';

// third-party libraries
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

// components
import MenuItems from '../MenuItems';

describe('MainView component', () => {
  let wrapper;
  const props = {
    id: 1,
    name: 'cat'
  }

  beforeEach(() => {
    wrapper = shallow(
      <MenuItems { ...props } />
    );
  });

  it('renders correctly', () => {
    const tree = toJson(wrapper);
    expect(tree).toMatchSnapshot();
  });
});
