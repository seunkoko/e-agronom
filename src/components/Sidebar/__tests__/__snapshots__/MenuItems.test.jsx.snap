// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainView component renders correctly 1`] = `
<Link
  className="links"
  to="/1"
>
  <div
    className="menu-items"
  >
    <i
      className="fas fa-cat menu-cat-icon"
    />
    <p
      className="menu-item-name"
    >
      cat
    </p>
  </div>
</Link>
`;
